### ISSUE TITLE
There maybe a bug in the backward of softplus

### ISSUE THREAD

**Issue by BBuf** [pytorch/pytorch#61568]
## 🐛 Bug

There maybe a bug in the backward of softplus

## To Reproduce

```
import torch
input = torch.from_numpy(np.array([0.2736]))
input.requires_grad = True
out = torch.nn.Softplus(beta=3, threshold=1)(input)
out = out.sum()
out.backward()
print(input.grad.numpy().tolist())
```

## Expected behavior

The expected gradient for input is 0.69440615, compared with 1.0 in the above program。

I tried the same program in PaddlePaddle and OneFlow, and the gradient is 0.69440615。

So I think the implementation here may have a bug, which does not conform to the gradient derived from the Softplus formula 

## Pytorch Version

1.8.1

cc @ezyang @albanD @zou3519 @gqchen @pearu @nikitaved @soulitzer @Lezcano @Varal7 @mruberry @jbschlosser

**Comment by H-Huang** [comment]
I can't repro this in the nightly build. I get the correct `0.694406` value. Perhaps this has been fixed?

**Comment by jbschlosser** [comment]
Is this the same problem as #55587? If so, it was fixed by #56484 and is in 1.9.
